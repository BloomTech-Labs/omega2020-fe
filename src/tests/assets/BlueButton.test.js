import React from 'react';
import { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Button from '../../components/assets/BlueButton';
import Adapter from 'enzyme-adapter-react-16';
import { Typography } from '@material-ui/core';

configure({ adapter: new Adapter() });

describe('the blue button content', () => {
  const mockInfo = {
    title: 'GitHub',
    href: 'https://github.com/Lambda-School-Labs/omega2020-fe',
  };

  const ReactWrapper = mount(<Button mockInfo={mockInfo}/>);
  const instanceWrapper = ReactWrapper.instance()

  it('snapshot the components', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('shows the instance to be null since there"s no state', () => {
    expect(instanceWrapper).toBe(null)
  })

  it('renders the mock info, and the length of the typography', () => {
    expect(ReactWrapper.find(Typography)).toHaveLength(1);
    expect(ReactWrapper.props().mockInfo.title).toEqual(mockInfo.title)
    expect(ReactWrapper.props().mockInfo.href).toEqual(mockInfo.href)

  });
});
