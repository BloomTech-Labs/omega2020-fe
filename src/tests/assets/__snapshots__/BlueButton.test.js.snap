// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the blue button content snapshot the components 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-text href MuiButton-textSizeLarge MuiButton-sizeLarge"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <p
      className="MuiTypography-root title MuiTypography-body2"
    />
  </span>
</button>
`;
